### DISCUSSION TITLE
[Kavita] Offset double page spreads

### DISCUSSION THREAD

**Discussion by morning-paradigm** [Kareadita/Kavita#2660] [Ideas]
### Idea Description

Just copypasting from https://github.com/Kareadita/Kavita/issues/1685 since the OP didn't want to do what they were asked to:

> As already discussed on discord, many scanlated chapters do not have a cover image and start directly at the first page. Mangadex's webreader gives an option to start the chapter at either page01 or page 01+page02 by having the option "offset double page spreads".
> 
> Komga and Cdisplayex have also the options to toggle the display mode to either with or no cover.
> 
> This allows chapters without a cover image to display properly in double page mode.
> 
> To add a little bit more context, manga chapters are designed with a white space in the middle, so that they can be bound to volumes. As is now, if a chapter does no have a cover image, it is being displayed incorrectly like so in Kavita:
> 
> https://i.imgur.com/7ivMbZE.png
> 
> And this is how the pages would be properly displayed, as intended:
> 
> https://i.imgur.com/0dbGwK7.png
> 

Tbh I also think it's more of an issue rather than a feature request.
It's especially annoying coupled with the bug (https://github.com/Kareadita/Kavita/issues/2494, https://github.com/Kareadita/Kavita/issues/2478) where some pages would occasionally not load at all causing the entire volume's pages to be misaligned. Btw, this bug is not iOS specific, I get it quite often in the web version, although I don't know how to reproduce it.

### Duration of Using Kavita

About a year

### Idea Category

User Experience

### Agreement

- [X] I agree that this is solely for submitting ideas, and I will search for existing ideas before posting.

**Comment by scare376** [comment]
Bumping as this is a rather common ask and many people create new discussions instead of commenting here.

**Comment by scare376** [comment]
Also to clarify why this isn't being considered an issue is because the double page mode is working as intended in this case. This is requesting additional functionality to an existing feature.

**Reply by rikuaokiTW** [reply]
I really have to push back on this being considered just 'additional functionality.' From a manga reading perspective, the absence of an offset feature can really impact the experience. While Kavita works wonderfully for many things, I believe this is a significant gap that prevents it from consistently displaying all manga properly, especially popular series like One Piece, JoJo and apparently Berserk too.

Using an example: I tested this with a .cbz file that closely matches the official digital color version of ONE PIECE Volume #1 (can be found online at Amazon Japan). As is common with manga volumes, it includes both a main volume cover and an internal chapter cover (It also includes a summary and some extra art pages, but even without those, the core issue remains, since the number would remain even). Because of this standard structure, Kavita's current handling of the very first page can throw off all the intended double-page spreads throughout the rest of the volume.

### **How Kavita displays the manga currently:**
<img width="1920" height="1080" alt="kavita-1" src="https://github.com/user-attachments/assets/71844d39-bbd2-43b1-81d6-8f7d9646aba8" />
<img width="1920" height="1080" alt="kavita-2" src="https://github.com/user-attachments/assets/5da74d79-ab36-41d0-a59d-8a67dc91727d" />
<img width="1920" height="1080" alt="kavita-3" src="https://github.com/user-attachments/assets/2e76bdc0-c763-4df1-abc8-3dec218e12d2" />
<img width="1920" height="1080" alt="kavita-4" src="https://github.com/user-attachments/assets/015e1735-707b-4c8d-9b9d-bb59ef6d51f6" />

### **The correct way to display it:**
<img width="1920" height="1080" alt="correct-1" src="https://github.com/user-attachments/assets/b64425f2-b175-40fb-b48a-6ca37ce02b28" />
<img width="1920" height="1080" alt="correct-2" src="https://github.com/user-attachments/assets/c0396f12-81ad-402b-92af-1b86d2c8eaed" />
<img width="1920" height="1080" alt="correct-3" src="https://github.com/user-attachments/assets/683a3de4-2a37-4ef3-9929-33e7ad1464e6" />


### **Physical volume reference:**
<img width="800" height="800" alt="physical" src="https://github.com/user-attachments/assets/68b76589-8ab6-4b80-b158-e957ec181eec" />

(OBS: I think it's better not to put the Amazon link, but if you look for the official version there, you'll see in the preview that the images that should be correctly distributed in 2 pages are like 5-6, 7-8, anyway, it's only possible if the cover is considered as page 1 and displayed with the 1-2 configuration, or maybe displaying the first and second image alone, since it's an even number, it's the way the manga is officially distributed)

This alignment issue isn't unique to full volumes either. For isolated chapters, like Chapter 1 of One Piece, which features a special color cover designed as a double page spread, the lack of this functionality means Kavita can't display it as intended. I've observed similar behaviors with Naruto volumes/chapters and others that follow what I consider a standard structure, and I'd be happy to provide more examples if that's helpful.

My core point is this: If Kavita aims to provide a great manga reading experience (and the 'double (manga)' display option certainly suggests it does), then the inability to correctly render these common manga formats becomes a notable limitation. It's not about poorly formatted files; it's about accommodating the standard way many digital manga are structured. I genuinely enjoy using Kavita for my EPUBs and PDFs, and this is the main challenge I've encountered with manga so far. I'm exploring potential solutions for my own use and hope this perspective can contribute to a future enhancement for all users.

**Reply by scare376** [reply]
The reason this is considered additional funtionality is because the feature was delivered as asked. A double page mode. The original function was for comics. the page order was changed to allow manga to function (hence the manga one was created). Iirc at the time nobody brought up offset and thus it wasnt implemented. the feature was shipped and thus having an offset is now considered additional functionality. Realistically if you have full volumes you shouldn't have an issue as additional pages are included to account for the offset as the cover isnt usually included in the page count.

**Reply by rikuaokiTW** [reply]
Hmmmmmmm, I see. Manga is something that comes later. I didn't know that. It makes more sense now. Thanks for the information. Sorry for any misconceptions in my previous comment about that.

And yes, considering the physical volumes, you're right: the first image you see inside is usually alone, since the main cover is on the outside, and the chapter usually starts on the back of that first page. But digital is different. As I mentioned, you can check the structure on the official websites through the free preview. Additionally, in Reddit communities, like the Kindle one, you can find similar threads due to some limitations of their app and volumes that deviate from the standard.

Anyway, I understand now that manga was more of a secondary development. Kavita works very well for the other types, and I'm grateful it exists, so I think everything's okay. I apologize for any inconvenience.

**Reply by scare376** [reply]
I want to say I'm not trying to downplay the importance of the feature I'm just trying to explain why we don't consider it to be an issue with the software that it's missing.
The way that I have been reading lately hasn't allowed me to enjoy the full page spreads plus most if not all of the digital manga that I have been reading include that offset page otherwise I'd probably be bugging Joe about it as well.

**Reply by majora2007** [reply]
Actually, Kavita was built for Manga only. But this FR was for Comic users, so as scare explained, when it was built and tested, we got feedback to add it for Manga and modeled it based on existing software (Komga) and everyone was happy. Obviously, loose chapters need an offset. Hence why there is a FR and with 26 votes, it's on my radar to slot it into a release. 

Kavita has been traditionally coded mainly by myself, so time is always limited. I try to balance bringing my visions and the collective feedback from the community to the software.

**Comment by majora2007** [comment]
This has been implemented as of v0.8.8.19 (nightly) and will be apart of the v0.8.9 (stable) release. Thanks @linkion for the PR.

**Comment by linkion** [comment]
Your welcome guys!
